extension/intents/forms/formsContentScript.js

```diff
@@ -233,4 +256,19 @@ this.dictationContentScript = (function() {
     // think of as "editable"
     focus(elements[0]);
   }
+
+  function replaceSelectedText(replacementText) {
+    let sel, range;
+    if (window.getSelection) {
+      sel = window.getSelection();
+      if (sel.rangeCount) {
+        range = sel.getRangeAt(0);
+        range.deleteContents();
+        range.insertNode(document.createTextNode(replacementText));
```

Comment:
This should be doing something like:

```js
const anchor = document.createElement("a");
anchor.textContent = title;
anchor.href = url;
range.insertNode(anchor);
```

Then it will insert a new rich link.